shards/shards.go

```diff
@@ -64,11 +64,14 @@ var (
 		Name: "zoekt_search_failed_total",
 		Help: "The total number of search requests that failed",
 	})
-	metricSearchDuration = promauto.NewHistogram(prometheus.HistogramOpts{
-		Name:    "zoekt_search_duration_seconds",
-		Help:    "The duration a search request took in seconds",
-		Buckets: prometheus.DefBuckets, // DefBuckets good for service timings
-	})
+	metricSearchDuration = promauto.NewHistogramVec(
```

Comment:
I'm curious how prometheus handles changes to metric types. Maybe because they're both histograms, it's not really a change and now there will just be labels?

Reply:
@jtibshirani 

In Prometheus' [data model](https://training.promlabs.com/training/introduction-to-prometheus/prometheus-an-overview/time-series-data-model), time series are uniquely identified by their name (`zoekt_search_duration_seconds`) and any attached labels (`success`). In my interpretation, this means that adding a new label creates a _new_ time series. When we write Grafana dasbhoards that query on `zoekt_search_duration_seconds{success=true}`,  the old data points (from `zoekt_search_duration_seconds` observations that occurred _before_ we added the `success` label ) won't be returned from the Prometheus server.

Does this answer your question?

Reply:
Yep, thanks for the reference.